limebit/medmodels
# refactor: sphinx setup

This PR introduces the sphinx documentation setup. It's a fairly large PR. 

# The following things changed: 

* `/docs/` directory contains the full sphinx & autodoc setup
* the `/image/` directory got removed. New repo with static medmodels media is available [here](https://github.com/limebit/medmodels-static)
* various corrections of faulty docstrings throughout medmodels
* `pyproject.toml` updated to include the dependencies for building the docs
* `Makefile` updated to include new targets: `make install-docs`, `make docs`, `make docs-serve` (live update & serve docs locally) and `make docs-clean`. 
  * docs related targets call the Makefile targets of `/docs/Makefile`

----

# How to test this PR locally

1. checkout branch 
2. run `make docs-serve`
3. go to http://localhost:5500/ and check the docs
4. Visit the documentation guide at http://localhost:5500/developer_guide/docstrings.html and provide feedback :) 

Thanks <3
